src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/CreateCaseTest.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.hmcts.ccd.datastore.tests.v2.external;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.ccd.datastore.tests.AATHelper;
+import uk.gov.hmcts.ccd.datastore.tests.BaseTest;
+import uk.gov.hmcts.ccd.datastore.tests.fixture.AATCaseBuilder;
+import uk.gov.hmcts.ccd.datastore.tests.fixture.AATCaseBuilder.FullCase;
+import uk.gov.hmcts.ccd.datastore.tests.fixture.AATCaseType;
+import uk.gov.hmcts.ccd.datastore.tests.fixture.AATCaseType.Event;
+import uk.gov.hmcts.ccd.domain.model.std.CaseDataContent;
+import uk.gov.hmcts.ccd.v2.V2;
+
+import java.util.function.Supplier;
+
+import static java.lang.Boolean.FALSE;
+import static org.hamcrest.Matchers.equalTo;
+import static uk.gov.hmcts.ccd.datastore.tests.fixture.AATCaseType.CASE_TYPE;
+import static uk.gov.hmcts.ccd.datastore.tests.fixture.AATCaseType.Event.CREATE;
+import static uk.gov.hmcts.ccd.datastore.tests.fixture.AATCaseType.JURISDICTION;
+
+@DisplayName("Create case")
+class CreateCaseTest extends BaseTest {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final String NOT_FOUND_CASE_REFERENCE = "1234123412341238";
+    private static final String INVALID_CASE_REFERENCE = "1234123412341234";
+    private static final String INVALID_EVENT_TRIGGER_ID = "invalidEvent";
+    private static final String EVENT_TOKEN = "someToken";
+
+    protected CreateCaseTest(AATHelper aat) {
+        super(aat);
+    }
+
+    @Test
+    @DisplayName("should create case")
+    void shouldCreateCase() {
+        String eventToken = aat.getCcdHelper().generateTokenCreateCase(asAutoTestCaseworker(), JURISDICTION, CASE_TYPE, CREATE);
+
+        callCreateCase(CASE_TYPE, getBody(CREATE, eventToken, FullCase::build))
+            .when()
+            .post("/case-types/{caseTypeId}/cases")
```

Comment:
I can see plenty of API responses defined for POST /case-types/{caseTypeId}/cases, but no 201. Maybe it is worth to test other responses. Please check.

Reply:
But this tests actually tests for 201. The functional tests are here to test mainly a successful scenario and some of failure ones. All failure ones are tested in lower levels.